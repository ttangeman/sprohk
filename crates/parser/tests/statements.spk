fn foo(x: i32) -> i32 {
    var y: u32 = 42;
}