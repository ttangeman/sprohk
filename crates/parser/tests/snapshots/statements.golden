node 0 {
  kind: TypeExpr
  tokens: [5..6]
  data: {
    root: 'i32'
  }
}
node 1 {
  kind: FnParameter
  tokens: [3..6]
  data: {
    name: x
    type_expr: 0
  }
}
node 2 {
  kind: TypeExpr
  tokens: [8..9]
  data: {
    root: 'i32'
  }
}
node 3 {
  kind: FnPrototype
  tokens: [0..9]
  data: {
    name: foo
    ret_type_expr: 2
  }
}
node 4 {
  kind: TypeExpr
  tokens: [13..14]
  data: {
    root: 'u32'
  }
}
node 5 {
  kind: AssignExpr
  tokens: [15..16]
  data: {
    literal: '42'
  }
}
node 6 {
  kind: VarDecl
  tokens: [10..17]
  data: {
    specifier: Var
    name: 'y'
    type_expr: 4
    assign_expr: 5
  }
}
node 7 {
  kind: Block
  tokens: [9..18]
  data: {
    statements: [
      6
    ]
  }
}
node 8 {
  kind: Function
  tokens: [0..18]
  data: {
    prototype: 3
    block: 7
  }
}
